src/audio/pipeline.c

```diff
@@ -750,15 +750,13 @@ static int pipeline_xrun_recover(struct pipeline *p)
 /* notify pipeline that this component requires buffers emptied/filled */
 void pipeline_schedule_copy(struct pipeline *p, uint64_t start)
 {
-	if (p->sched_comp->state == COMP_STATE_ACTIVE) {
+	if (p->sched_comp->state == COMP_STATE_ACTIVE)
+#ifdef CONFIG_TIMER_SCHEDULING
 		/* timer driven pipeline should execute task synchronously */
-		if (p->ipc_pipe.timer_delay)
-			pipeline_copy(p->sched_comp);
-		else
-			schedule_task(&p->pipe_task, start,
-				      p->ipc_pipe.deadline);
-	}
-
+		pipeline_copy(p->sched_comp);
```

Comment:
Whether the pipeline is timer or IRQ driver should be configurable at runtime.  The pipeline IPC data should set the type and then pipeline runtime should manage appropriately. I dont see any change here from today since it handle both cases.

Reply:
Today it's unchanged, but I'm working on huge changeset, that will modify the pipeline processing flow. I don't think it's a good idea to have it configured at runtime. With heavy DSP processing and many pipelines it will become almost unmanageable. It means mixing work_queue processing (with priorities in the future) and scheduler processing and many different irq levels in the same time.

Reply:
WorkQ and scheduler are now both doing the same thing.... The workQ came first and was intended for performing small work loads on regular intervals. Then came the scheduler which was more powerful/complex and overlaps somewhat with workQ.

It would make sense to me now to deduplicate and remove the workQ. Lets perform all work/tasks via the scheduler. This keeps the footprint smaller and code simpler.